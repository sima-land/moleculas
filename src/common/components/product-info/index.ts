export { type ProductInfoProps, ProductInfo, Parts } from './product-info';
