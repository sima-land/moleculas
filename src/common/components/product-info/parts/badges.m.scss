.badges {
  margin-bottom: 8px;
}
