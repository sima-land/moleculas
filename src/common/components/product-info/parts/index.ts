import { ProductImage } from './image';
import { ImageButton } from './image-button';
import { ProductInfoBadges } from './badges';
import { ProductInfoPrices } from './prices';
import { ProductInfoTitle } from './title';
import { ProductInfoSecondaryInfo } from './secondary-info';
import { ProductInfoTrademarkLink } from './trademark-link';
import { ProductInfoRatingCounter } from './rating-couter';
import { ProductInfoFooter } from './footer';
import { ProductInfoFooterContext } from '../utils';
import { ProductInfoCartControl } from './cart-control';
import { WaitListAddedLink } from './wait-list-added-link';
import { WaitListAddLink } from './wait-list-add-link';
import { AdultConfirmButton } from './adult-confirm-button';

export const Parts = {
  // основные компоненты-слоты и компоненты-запчасти
  Image: ProductImage,
  ImageButton,
  Badges: ProductInfoBadges,
  Prices: ProductInfoPrices,
  Title: ProductInfoTitle,
  SecondaryInfo: ProductInfoSecondaryInfo,
  TrademarkLink: ProductInfoTrademarkLink,
  RatingCounter: ProductInfoRatingCounter,

  // компоненты-слоты и компоненты-запчасти, предназначенные для вывода футера
  Footer: ProductInfoFooter,
  FooterContext: ProductInfoFooterContext,
  CartControl: ProductInfoCartControl,
  AdultConfirmButton,
  WaitListAddButton: WaitListAddLink,
  WaitListAddedLink,
} as const;
