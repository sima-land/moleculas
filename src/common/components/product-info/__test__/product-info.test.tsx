import { render, fireEvent } from '@testing-library/react';
import { ProductInfo } from '../product-info';
import { Parts } from '../parts';
import { Badge, BadgeProps } from '../../badge';
import QuickViewSVG from '@sima-land/ui-quarks/icons/24x24/Stroked/MagnifierPlus';
import { Button } from '@sima-land/ui-nucleons/button';
import { Stepper } from '@sima-land/ui-nucleons/stepper';

const normalizePrice = (s: string): string => s.replace(/\s+/g, '').replace(',', '.');

const badges: BadgeProps[] = [
  { color: '#b52ea8', fields: [{ type: 'text', value: '-56%' }] },
  { color: '#ff7200', fields: [{ type: 'text', value: '3 по цене 2' }] },
  { color: '#2962ff', fields: [{ type: 'text', value: 'Товар месяца' }] },
  { color: '#00b8d4', fields: [{ type: 'text', value: 'Акция' }] },
];

const data = {
  name: 'Some product',
  url: 'https://www.products.com/some-product',
  imageSrc: 'https://www.images.com/some-product/1',
  price: 23476.23,
  oldPrice: 25000,
  currencyGrapheme: '$',
  trademark: { name: 'Test Trademark', url: 'https://www.test-trademark.ru/' },
  badges,
  cart: {
    stepInfo: 'По 7 шт',
    markupInfo: 'Комплектация + 50 ₽ при покупке до 20 шт',
  },
} as const;

describe('ProductInfo', () => {
  describe('regular content', () => {
    const spies = {
      imageClick: jest.fn(),
      quickViewClick: jest.fn(),
      badgeClick: jest.fn(),
      titleClick: jest.fn(),
      trademarkClick: jest.fn(),
    };

    const template = (
      <ProductInfo>
        <Parts.Image href={data.url} src={data.imageSrc} onClick={spies.imageClick}>
          <Parts.ImageButton
            icon={QuickViewSVG}
            onClick={spies.quickViewClick}
            data-testid='test-favorite-button'
            hint='Добавить в избранное'
          />
          <Parts.ImageButton
            icon={QuickViewSVG}
            onClick={spies.quickViewClick}
            data-testid='quick-view-button'
          />
        </Parts.Image>

        <Parts.Badges>
          {data.badges.map((badge, index) => (
            <Badge key={index} {...badge} onClick={() => spies.badgeClick(badge)} />
          ))}
        </Parts.Badges>

        <Parts.Prices
          price={data.price}
          oldPrice={data.oldPrice}
          currencyGrapheme={data.currencyGrapheme}
        />

        <Parts.Title href={data.url} onClick={spies.titleClick}>
          {data.name}
        </Parts.Title>

        <Parts.SecondaryInfo>Арт. 123456</Parts.SecondaryInfo>

        <Parts.TrademarkLink href={data.trademark.url} onClick={spies.trademarkClick}>
          {data.trademark.name}
        </Parts.TrademarkLink>

        <Parts.RatingCounter value={4.8} reviewCount={943} />

        <Parts.Footer>
          <Parts.CartControl stepText={data.cart.stepInfo} markupText={data.cart.markupInfo}>
            <Button size='s' data-testid='test-add-to-cart-button'>
              В корзину
            </Button>
          </Parts.CartControl>
        </Parts.Footer>
      </ProductInfo>
    );

    it('should match snapshot', () => {
      const { container } = render(template);

      expect(container).toMatchSnapshot();

      expect(spies.imageClick).toHaveBeenCalledTimes(0);
      expect(spies.quickViewClick).toHaveBeenCalledTimes(0);
      expect(spies.badgeClick).toHaveBeenCalledTimes(0);
      expect(spies.titleClick).toHaveBeenCalledTimes(0);
      expect(spies.trademarkClick).toHaveBeenCalledTimes(0);
    });

    it('image-link click', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:image-link').getAttribute('href')).toBe(data.url);
      expect(spies.imageClick).toHaveBeenCalledTimes(0);
      fireEvent.click(getByTestId('product-info:image-link'));
      expect(spies.imageClick).toHaveBeenCalledTimes(1);
    });

    it('image icon-button click', () => {
      const { getByTestId } = render(template);

      expect(spies.quickViewClick).toHaveBeenCalledTimes(0);
      fireEvent.click(getByTestId('quick-view-button'));
      expect(spies.quickViewClick).toHaveBeenCalledTimes(1);
    });

    it('positioned image icon-buttons', () => {
      const { container } = render(
        <ProductInfo>
          <Parts.Image href={data.url} src={data.imageSrc} onClick={spies.imageClick}>
            <Parts.ImageButton
              icon={QuickViewSVG}
              position={{ x: 'left', y: 'top' }}
              onClick={spies.quickViewClick}
              data-testid='test-favorite-button'
              hint='Добавить в избранное'
            />
            <Parts.ImageButton
              icon={QuickViewSVG}
              position={{ x: 'right', y: 'bottom' }}
              onClick={spies.quickViewClick}
              data-testid='quick-view-button'
            />
          </Parts.Image>

          <Parts.Prices
            price={data.price}
            oldPrice={data.oldPrice}
            currencyGrapheme={data.currencyGrapheme}
          />

          <Parts.Title href={data.url} onClick={spies.titleClick}>
            {data.name}
          </Parts.Title>
        </ProductInfo>,
      );

      expect(container).toMatchSnapshot();
    });

    it('image icon-button hint', async () => {
      const { getByTestId, findByTestId, container } = render(template);

      fireEvent.mouseEnter(getByTestId('test-favorite-button'));
      expect((await findByTestId('hint')).textContent).toContain('Добавить в избранное');

      fireEvent.mouseLeave(getByTestId('test-favorite-button'));
      expect(container.querySelectorAll('[data-testid="hint"]')).toHaveLength(0);
    });

    it('image src', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:image').getAttribute('src')).toBe(data.imageSrc);
    });

    it('badges', () => {
      const { getAllByTestId } = render(template);

      expect(data.badges).toHaveLength(4);
      expect(getAllByTestId('badge')).toHaveLength(data.badges.length);
    });

    it('badges click', () => {
      const { getAllByTestId } = render(template);

      expect(spies.badgeClick).toHaveBeenCalledTimes(0);
      fireEvent.click(getAllByTestId('badge')[data.badges.length - 1]);
      expect(spies.badgeClick).toHaveBeenCalledTimes(1);
      expect(spies.badgeClick).toHaveBeenCalledWith(data.badges[data.badges.length - 1]);
    });

    it('prices', () => {
      const { getByTestId } = render(template);

      const price = normalizePrice(getByTestId('product-info:price').textContent || '');
      const oldPrice = normalizePrice(getByTestId('product-info:old-price').textContent || '');

      expect(price).toContain(String(data.price));
      expect(price).toContain(data.currencyGrapheme);

      expect(oldPrice).toContain(String(data.oldPrice));
      expect(oldPrice).toContain(data.currencyGrapheme);
    });

    it('title', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:name-link').textContent).toContain(data.name);
      expect(getByTestId('product-info:name-link').getAttribute('href')).toBe(data.url);
      expect(spies.titleClick).toHaveBeenCalledTimes(0);
      fireEvent.click(getByTestId('product-info:name-link'));
      expect(spies.titleClick).toHaveBeenCalledTimes(1);
    });

    it('trademark', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:trademark-link').textContent).toContain(data.trademark.name);
      expect(spies.trademarkClick).toHaveBeenCalledTimes(0);
      fireEvent.click(getByTestId('product-info:trademark-link'));
      expect(spies.trademarkClick).toHaveBeenCalledTimes(1);
    });

    it('cart control in footer', () => {
      const { getAllByTestId, getByTestId } = render(template);

      expect(getAllByTestId('test-add-to-cart-button')).toHaveLength(1);
      expect(getByTestId('product-info:cart-step').textContent).toContain(data.cart.stepInfo);
      expect(getByTestId('product-info:cart-info').textContent).toContain(data.cart.markupInfo);
    });
  });

  describe('adult content', () => {
    const spies = {
      adultConfirm: jest.fn(),
    };

    const template = (
      <ProductInfo restriction='adult'>
        <Parts.Image src={data.imageSrc} href={data.url}>
          <Parts.ImageButton icon={QuickViewSVG} data-testid='quick-view-button' />
        </Parts.Image>

        <Parts.Badges>
          {data.badges.map((badge, index) => (
            <Badge key={index} {...badge} />
          ))}
        </Parts.Badges>

        <Parts.Prices
          price={data.price}
          oldPrice={data.oldPrice}
          currencyGrapheme={data.currencyGrapheme}
        />

        <Parts.Title href={data.url}>{data.name}</Parts.Title>

        <Parts.TrademarkLink href={data.trademark.url}>{data.trademark.name}</Parts.TrademarkLink>

        <Parts.Footer>
          <Parts.AdultConfirmButton onClick={spies.adultConfirm} />
        </Parts.Footer>
      </ProductInfo>
    );

    it('should match snapshot', () => {
      const { container } = render(template);

      expect(container).toMatchSnapshot();
    });

    it('should not render image-link', () => {
      const { container } = render(template);

      expect(container.querySelectorAll('[data-testid="product-info:image-link"]')).toHaveLength(0);
    });

    it('should render adult image', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:adult-image').getAttribute('src')).toBe(data.imageSrc);
    });

    it('should hide image icon-button', () => {
      const { container } = render(template);

      expect(container.querySelectorAll('[data-testid="quick-view-button"]')).toHaveLength(0);
    });

    it('should not render badges', () => {
      const { container } = render(template);

      expect(container.querySelectorAll('[data-testid="badge"]')).toHaveLength(0);
    });

    it('should render only actual price but not old price', () => {
      const { container, getByTestId } = render(template);

      const price = normalizePrice(getByTestId('product-info:price').textContent || '');

      expect(price).toContain(String(data.price));
      expect(price).toContain(data.currencyGrapheme);

      expect(container.querySelectorAll('[data-testid="product-info:old-price"]')).toHaveLength(0);
    });

    it('should replace title by adult warning', () => {
      const { container, getByTestId } = render(template);

      expect(container.querySelectorAll('[data-testid="product-info:name-link"]')).toHaveLength(0);
      expect(getByTestId('product-info:adult-warning').textContent).toContain(
        'Подтвердите, что вы старше 18 лет, чтобы посмотреть товар.',
      );
    });

    it('should not render trademark', () => {
      const { container } = render(template);

      expect(
        container.querySelectorAll('[data-testid="product-info:trademark-link"]'),
      ).toHaveLength(0);
    });

    it('should render adult confirm button', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:adult-confirm-button').textContent).toBe('Подтвердить');
      expect(spies.adultConfirm).toHaveBeenCalledTimes(0);
      fireEvent.click(getByTestId('product-info:adult-confirm-button'));
      expect(spies.adultConfirm).toHaveBeenCalledTimes(1);
    });
  });

  describe('unavailable content', () => {
    const spies = {
      addToWaitList: jest.fn(),
      waitListLinkClick: jest.fn(),
    };

    const template = (
      <ProductInfo restriction='unavailable'>
        <Parts.Image src={data.imageSrc} href={data.url} opacity={0.72}>
          <Parts.ImageButton icon={QuickViewSVG} data-testid='quick-view-button' />
        </Parts.Image>

        <Parts.Badges>
          {data.badges.map((badge, index) => (
            <Badge key={index} {...badge} />
          ))}
        </Parts.Badges>

        <Parts.Prices
          price={data.price}
          oldPrice={data.oldPrice}
          currencyGrapheme={data.currencyGrapheme}
          unavailableReason='Test reason'
        />

        <Parts.Title href={data.url}>{data.name}</Parts.Title>

        <Parts.TrademarkLink href={data.trademark.url}>{data.trademark.name}</Parts.TrademarkLink>

        <Parts.Footer>
          <Parts.WaitListAddButton onClick={spies.addToWaitList} />
          <Parts.WaitListAddedLink href='https://www.wait.com/' onClick={spies.waitListLinkClick} />
        </Parts.Footer>
      </ProductInfo>
    );

    it('should handle image opacity', () => {
      const { getByTestId } = render(template);

      expect(getByTestId('product-info:image').style.opacity).toBe('0.72');
    });

    it('badges', () => {
      const { getAllByTestId } = render(template);

      expect(data.badges).toHaveLength(4);
      expect(getAllByTestId('badge')).toHaveLength(data.badges.length);
    });

    it('should not render prices but unavailable reason', () => {
      const { container, getByTestId } = render(template);

      expect(container.querySelectorAll('[data-testid="product-info:price"]')).toHaveLength(0);
      expect(container.querySelectorAll('[data-testid="product-info:old-price"]')).toHaveLength(0);
      expect(getByTestId('product-info:prices').textContent).toContain('Test reason');
    });
  });

  it('should render stub for broken image', () => {
    const { container, getByTestId } = render(
      <ProductInfo>
        <Parts.Image src={data.imageSrc} href={data.url} opacity={0.72}>
          <Parts.ImageButton icon={QuickViewSVG} data-testid='quick-view-button' />
        </Parts.Image>

        <Parts.Prices
          price={data.price}
          oldPrice={data.oldPrice}
          currencyGrapheme={data.currencyGrapheme}
          unavailableReason='Test reason'
        />

        <Parts.Title href={data.url}>{data.name}</Parts.Title>
      </ProductInfo>,
    );

    expect(container.querySelectorAll('.broken-icon')).toHaveLength(0);
    fireEvent.error(getByTestId('product-info:image'));
    expect(container.querySelectorAll('.broken-icon')).toHaveLength(1);
  });

  it('should render stub for broken adult image', () => {
    const { container, getByTestId } = render(
      <ProductInfo restriction='adult'>
        <Parts.Image src={data.imageSrc} href={data.url} opacity={0.72}>
          <Parts.ImageButton icon={QuickViewSVG} data-testid='quick-view-button' />
        </Parts.Image>

        <Parts.Prices
          price={data.price}
          oldPrice={data.oldPrice}
          currencyGrapheme={data.currencyGrapheme}
          unavailableReason='Test reason'
        />

        <Parts.Title href={data.url}>{data.name}</Parts.Title>
      </ProductInfo>,
    );

    expect(container.querySelectorAll('.broken-icon')).toHaveLength(0);
    fireEvent.error(getByTestId('product-info:adult-image'));
    expect(container.querySelectorAll('.broken-icon')).toHaveLength(0);
    expect(getByTestId('product-info:adult-image').classList.contains('broken')).toBe(true);
  });

  it('should allow place random children when strict is false', () => {
    const { container } = render(
      <ProductInfo strict={false}>
        <Parts.Title href='https://my-site.com'>My product!</Parts.Title>
        <Parts.Image src='https://my-site.com/images/1' href='https://my-site.com' />
        <div>My custom child element!</div>
      </ProductInfo>,
    );

    expect(container.textContent).toBe('My product!My custom child element!');
  });

  it('should render loading cart control', () => {
    const { container, rerender } = render(
      <ProductInfo strict={false}>
        <Parts.Image src='https://my-site.com/images/1' href='https://my-site.com' />
        <Parts.Title href='https://my-site.com'>My product!</Parts.Title>
        <Parts.Footer>
          <Parts.CartControl
            loading
            stepText='По 5 шт'
            markupText='Комплектация + 50 ₽ при покупке до 20 шт'
          >
            <Stepper defaultValue={5} size='s' />
          </Parts.CartControl>
        </Parts.Footer>
      </ProductInfo>,
    );

    expect(container.textContent).toBe('My product!');

    rerender(
      <ProductInfo strict={false}>
        <Parts.Image src='https://my-site.com/images/1' href='https://my-site.com' />
        <Parts.Title href='https://my-site.com'>My product!</Parts.Title>
        <Parts.Footer>
          <Parts.CartControl
            loading={false}
            stepText='По 5 шт'
            markupText='Комплектация + 50 ₽ при покупке до 20 шт'
          >
            <Stepper defaultValue={5} size='s' />
          </Parts.CartControl>
        </Parts.Footer>
      </ProductInfo>,
    );

    expect(container.textContent).toBe(
      'My product!По 5 штКомплектация + 50 ₽ при покупке до 20 шт',
    );
  });
});
