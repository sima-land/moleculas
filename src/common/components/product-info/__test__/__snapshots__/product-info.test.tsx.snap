// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfo adult content should match snapshot 1`] = `
<div>
  <span
    class="root image-overlay"
  >
    <img
      class="image adult"
      data-testid="product-info:adult-image"
      src="https://www.images.com/some-product/1"
    />
    <svg
      class="adult-icon"
      height="64"
      viewBox="0 0 64 64"
      width="64"
    >
      <path
        clip-rule="evenodd"
        d="M54.827 19.2h-2.56v-7.253h-7.254v-2.56h7.254V2.133h2.56v7.254h7.253v2.56h-7.253V19.2ZM36.27 27.224c.656.744 1.866 1.487 3.634 2.23 1.589-.675 2.713-1.4 3.374-2.178a4.07 4.07 0 0 0 .992-2.72c0-1.137-.402-2.041-1.208-2.712-.805-.67-1.914-1.004-3.324-1.004-1.39 0-2.478.337-3.267 1.013-.789.675-1.184 1.577-1.184 2.702 0 1.037.328 1.927.983 2.67Zm-.65 7.02c-.744.872-1.116 1.917-1.116 3.133 0 1.351.463 2.404 1.391 3.16.928.755 2.23 1.13 3.909 1.13 1.656 0 2.944-.392 3.865-1.18.923-.79 1.385-1.87 1.385-3.245 0-1.092-.433-2.063-1.3-2.913-.867-.851-2.378-1.676-4.533-2.476-1.656.72-2.856 1.518-3.6 2.391Zm-2.049-6.386c-.666-1.018-1-2.159-1-3.421 0-1.79.652-3.218 1.958-4.282 1.305-1.065 3.052-1.597 5.242-1.597 2.222 0 3.983.524 5.283 1.572 1.3 1.047 1.95 2.495 1.95 4.342 0 1.216-.372 2.325-1.117 3.328-.744 1.002-1.933 1.914-3.567 2.736 1.978.956 3.384 1.962 4.217 3.016.834 1.053 1.25 2.272 1.25 3.657 0 2.05-.706 3.685-2.116 4.907-1.412 1.222-3.345 1.833-5.8 1.833-2.6 0-4.6-.578-6-1.731-1.4-1.155-2.1-2.79-2.1-4.908 0-2.826 1.7-5.029 5.1-6.606-1.534-.877-2.634-1.826-3.3-2.846ZM32 64C14.327 64 0 49.673 0 32 0 14.327 14.327 0 32 0c3.73 0 7.313.638 10.642 1.812.025.568.025 2.677.025 2.668C39.347 3.183 35.779 2.56 32 2.56 15.74 2.56 2.56 15.74 2.56 32S15.74 61.44 32 61.44 61.44 48.26 61.44 32c0-3.769-.708-7.372-1.999-10.685 1.412.018 2.489.018 2.737.015A31.943 31.943 0 0 1 64 32c0 17.673-14.327 32-32 32ZM16.829 25.89l-1.466-1.925 6.45-5.052h2.332v24.698h-2.7V26.008c0-1.464.044-2.849.134-4.155-.234.236-.495.485-.784.744-.288.259-1.61 1.357-3.966 3.293Z"
        fill-rule="evenodd"
      />
    </svg>
  </span>
  <div
    class="prices height-limit"
    data-testid="product-info:prices"
  >
    <span
      class="root price"
      data-testid="product-info:price"
    >
      23 476,23 $
    </span>
  </div>
  <span
    class="adult-info"
    data-testid="product-info:adult-warning"
  >
    Подтвердите, что вы старше 18 лет, чтобы посмотреть товар.
  </span>
  <div
    class="footer"
    data-testid="product-info:footer"
  >
    <a
      class="link text-button basic-blue"
      data-testid="product-info:adult-confirm-button"
      role="button"
      tabindex="0"
    >
      Подтвердить
    </a>
  </div>
</div>
`;

exports[`ProductInfo regular content positioned image icon-buttons 1`] = `
<div>
  <span
    class="root image-overlay"
  >
    <a
      class="image-link"
      data-testid="product-info:image-link"
      href="https://www.products.com/some-product"
    >
      <img
        class="image"
        data-testid="product-info:image"
        src="https://www.images.com/some-product/1"
      />
    </a>
    <div
      class="image-buttons"
    >
      <div
        class="root image-button positioned left-top"
        data-testid="test-favorite-button"
        role="banner"
      >
        <svg
          class="stroke"
          height="24"
          stroke="#f5f5f5"
          stroke-width="3px"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
        <svg
          class="svg"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
      </div>
      <div
        class="root image-button positioned right-bottom"
        data-testid="quick-view-button"
        role="banner"
      >
        <svg
          class="stroke"
          height="24"
          stroke="#f5f5f5"
          stroke-width="3px"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
        <svg
          class="svg"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
      </div>
    </div>
  </span>
  <div
    class="prices height-limit"
    data-testid="product-info:prices"
  >
    <span
      class="root price"
      data-testid="product-info:price"
    >
      23 476,23 $
    </span>
    <span
      class="root old-price crossed-out"
      data-testid="product-info:old-price"
    >
      25 000 $
    </span>
  </div>
  <a
    class="link link line-clamp basic-gray87"
    data-testid="product-info:name-link"
    href="https://www.products.com/some-product"
  >
    Some product
  </a>
</div>
`;

exports[`ProductInfo regular content should match snapshot 1`] = `
<div>
  <span
    class="root image-overlay"
  >
    <a
      class="image-link"
      data-testid="product-info:image-link"
      href="https://www.products.com/some-product"
    >
      <img
        class="image"
        data-testid="product-info:image"
        src="https://www.images.com/some-product/1"
      />
    </a>
    <div
      class="image-buttons"
    >
      <div
        class="root image-button"
        data-testid="test-favorite-button"
        role="banner"
      >
        <svg
          class="stroke"
          height="24"
          stroke="#f5f5f5"
          stroke-width="3px"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
        <svg
          class="svg"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
      </div>
      <div
        class="root image-button"
        data-testid="quick-view-button"
        role="banner"
      >
        <svg
          class="stroke"
          height="24"
          stroke="#f5f5f5"
          stroke-width="3px"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
        <svg
          class="svg"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M3 11.5a8.5 8.5 0 1 1 17 0 8.5 8.5 0 0 1-17 0ZM11.5 5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13Z"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M16.293 16.293a1 1 0 0 1 1.414 0l4 4a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414Z"
            fill-rule="evenodd"
          />
          <path
            d="M10.5 9a1 1 0 1 1 2 0v5a1 1 0 1 1-2 0V9Z"
          />
          <path
            d="M9 12.5a1 1 0 1 1 0-2h5a1 1 0 1 1 0 2H9Z"
          />
        </svg>
      </div>
    </div>
  </span>
  <div
    class="root badges"
  >
    <div
      class="item"
    >
      <a
        class="root padding-x-default coloring-outline"
        data-testid="badge"
        style="--badge-color: #b52ea8;"
      >
        <span
          class="content"
        >
          -56%
        </span>
      </a>
    </div>
    <div
      class="item"
    >
      <a
        class="root padding-x-default coloring-outline"
        data-testid="badge"
        style="--badge-color: #ff7200;"
      >
        <span
          class="content"
        >
          3 по цене 2
        </span>
      </a>
    </div>
    <div
      class="item"
    >
      <a
        class="root padding-x-default coloring-outline"
        data-testid="badge"
        style="--badge-color: #2962ff;"
      >
        <span
          class="content"
        >
          Товар месяца
        </span>
      </a>
    </div>
    <div
      class="item"
    >
      <a
        class="root padding-x-default coloring-outline"
        data-testid="badge"
        style="--badge-color: #00b8d4;"
      >
        <span
          class="content"
        >
          Акция
        </span>
      </a>
    </div>
  </div>
  <div
    class="prices height-limit"
    data-testid="product-info:prices"
  >
    <span
      class="root price"
      data-testid="product-info:price"
    >
      23 476,23 $
    </span>
    <span
      class="root old-price crossed-out"
      data-testid="product-info:old-price"
    >
      25 000 $
    </span>
  </div>
  <a
    class="link link line-clamp basic-gray87"
    data-testid="product-info:name-link"
    href="https://www.products.com/some-product"
  >
    Some product
  </a>
  <div
    class="secondary-info"
  >
    Арт. 123456
  </div>
  <a
    class="link trademark-link basic-blue"
    data-testid="product-info:trademark-link"
    href="https://www.test-trademark.ru/"
  >
    Test Trademark
  </a>
  <div
    class="root size-unset hover-disabled rating"
    data-testid="rating-counter"
  >
    <svg
      class="star"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <path
        d="M7.536 1.545a.5.5 0 0 1 .928 0l1.711 4.262 4.583.311a.5.5 0 0 1 .287.883L11.52 9.945l1.12 4.454a.5.5 0 0 1-.75.546L8 12.503l-3.89 2.442a.5.5 0 0 1-.75-.546l1.12-4.454L.955 7.001a.5.5 0 0 1 .287-.883l4.583-.31 1.711-4.263Z"
      />
    </svg>
    <span
      class="value"
    >
      4,8
    </span>
    <span
      class="count"
    >
      943 отзыва
    </span>
  </div>
  <div
    class="footer"
    data-testid="product-info:footer"
  >
    <div
      class="control-wrapper"
    >
      <div
        class="control"
      >
        <button
          class="root size-s view-primary"
          data-testid="test-add-to-cart-button"
        >
          В корзину
        </button>
      </div>
      <div
        class="unit-text"
        data-testid="product-info:cart-step"
      >
        По 7 шт
      </div>
      <div
        class="bottom-text"
        data-testid="product-info:cart-info"
      >
        Комплектация + 50 ₽ при покупке до 20 шт
      </div>
    </div>
  </div>
</div>
`;
